R/zchunk_LA122.Refining_china.R

```diff
@@ -0,0 +1,238 @@
+   L101.inNBS_Mtce_province_S_F %>%
+      filter(sector == "refinery",
+             fuel == "crude oil") %>%
+      mutate(sector = "oil refining") %>%
+      replace_na(list(value = 0)) %>%
+      group_by(year) %>%
+      # province percentage of total in each year
+      mutate(value = value / sum(value, na.rm = T)) %>%
+      ungroup() ->
+      L122.net_Mtce_province_cor_crude
+
+    L122.net_Mtce_province_cor_crude %>%
+      group_by(province, sector, fuel) %>%
+      mutate(value = as.numeric(value), value = approx_fun(year, value, rule = 2)) %>%
+      ungroup() ->
+      L122.pct_province_cor
+
+    # Crude oil refining output by province
+    # Apportion the national total to the provinces
+    L122.pct_province_cor %>%
+      left_join_error_no_match(filter(L122.out_EJ_R_refining_F_Yh, GCAM_region_ID == 11 & sector == "oil refining"),
+                by = c("sector", "year")) %>%
+      # province output value = province proportion * national output value
+      mutate(value = value.x * value.y) %>%
+      select(province, sector, fuel = fuel.x, year, value) ->
+      L122.out_EJ_province_cor
+
+    # Inputs to crude oil refining - same method of portional allocations, but with multiple fuels
+    # Oil refining input fuels
+    L122.in_EJ_R_refining_F_Yh %>%
+      filter(sector == "oil refining") %>%
+      select(fuel) %>%
+      distinct() ->
+      oil_input_fuels
+
+    # Repeat province proportions for all fuels in oil refining sector
+    L122.pct_province_cor %>%
+      select(-fuel) %>%
+      repeat_add_columns(oil_input_fuels) ->
+      L122.pct_province_cor_repF
+
+    # Calculate province oil input values
+    L122.pct_province_cor_repF %>%
+      left_join_error_no_match(filter(L122.in_EJ_R_refining_F_Yh, GCAM_region_ID == 11), by = c("sector", "fuel", "year"))  %>%
+      # province input value = province proportion * national input value
+      mutate(value = value.x * value.y) %>%
+      select(province, sector, fuel, year, value) ->
+      L122.in_EJ_province_cor_F
+
+    # BIOMASS LIQUIDS
+    # Downscale ethanol production to province
+
+    # Calculate the percentages of corn ethanol consumption in each province
+    biofuel_MT_province_F %>%
+      gather(year, value, -province, -fuel) %>%
+      filter(fuel == "ethanol") %>%
+      mutate(year = as.numeric(year)) %>%
+      complete(nesting(fuel, province), year = HISTORICAL_YEARS) %>%
+      group_by(province, fuel) %>%
+      mutate( value = as.numeric(value), value = approx_fun(year, value, rule = 1)) %>%
+      ungroup() %>%
+      # ensure all provinces are in the data.frame even thugh they will just be NA
+      full_join(province_names_mappings["province"], by = "province") %>%
+      mutate(fuel = 'corn', sector = "corn ethanol") %>%
+      complete(sector, fuel, province, year = HISTORICAL_YEARS) %>%
+      group_by(year) %>%
+      mutate(value = value / sum(value, na.rm=T)) %>%
+      ungroup() %>%
+      replace_na(list(value = 0)) ->
```

Comment:
Please double check the old code, and make sure the NAs are replaced at the right step, even though it may not affect the values for now. It seems to me you'll not have NAs at this step because `sum(value, na.rm = T)`. If replace before the sum, then you don't need `na.rm = T`